pkg/fakerp/runcommand.go

```diff
@@ -0,0 +1,49 @@
+package fakerp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/go-chi/chi"
+)
+
+// handleRunCommand handles running generic commands on a given vm within a scaleset in the cluster
+func (s *Server) handleRunCommand(w http.ResponseWriter, req *http.Request) {
+	cs := s.read()
+	if cs == nil {
+		s.internalError(w, "Failed to read the internal config")
+		return
+	}
+
+	vm := chi.URLParam(req, "virtualMachine")
+	ss := chi.URLParam(req, "scaleSet")
+	command, err := readCommandInput(req)
+	if err != nil {
+		s.internalError(w, fmt.Sprintf("Failed to read command input: %v", err))
+		return
+	}
+
+	ctx := enrichContext(context.Background())
+	result, err := s.plugin.RunGenericCommand(ctx, cs, ss, vm, *command)
+	if err != nil {
+		s.internalError(w, fmt.Sprintf("Failed to run command: %v", err))
+		return
+	}
+
+	w.Write(result)
+	s.log.Infof("ran command on %s in %s", vm, ss)
+}
+
+func readCommandInput(req *http.Request) (*compute.RunCommandInput, error) {
+	data, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read request body: %v", err)
+	}
+	var params *compute.RunCommandInput
```

Comment:
How do you envision the RunCommandInput stuct working?  Do we have to create scripts to lay down or is the scripts section a bash string?

Reply:
It will be something like this:

```
func (u *simpleUpgrader) RestartDocker(ctx context.Context, oc *api.OpenShiftManagedCluster, scaleSetName, instanceId string) ([]byte, *api.PluginError) {
	params := compute.RunCommandInput{
		CommandID: to.StringPtr("RunShellScript"),
		Script: to.StringSlicePtr([]string{"systemctl restart docker.service"}),
	}
	return u.RunCommand(ctx, oc, scaleSetName, instanceId, params)
}
```

Reply:
Dont think we will have this luxury to lay down the script. I think this needs to be able to accept pasted in the script of some sort. Laying down means reimage. and if stuff is broken this mostly will not be an option.